notebooker/web/static/notebooker/index.js

```diff
@@ -1,34 +1,124 @@
+        path = path.substring(path.indexOf('/') + 1)
+        level = level - 1
+    }
+    return path
+}
+
+function updateContents(currentFolder, entries){
+    let $cardContainer = $('#cardContainer');
+    $cardContainer.empty();
+    subfoldersInfo = {}
+    reportParts = []
+    for (let report in entries) {
+        report_data = entries[report]
+        report_path = report_data.report_name
+        if(!report_path.startsWith(currentFolder)){
+            continue;
+        }
+        remainingPath = report_path.substring(currentFolder.length)
+        if(remainingPath.includes('/')) {
+            // it is a folder
+            subfolderPathName = remainingPath.substring(0, remainingPath.indexOf('/'))
+            subfolderPath = currentFolder + subfolderPathName
+            if (subfolderPath in subfoldersInfo) {
+                info.reportCount = info.reportCount + 1
+            } else{
+                info = {}
+                info.pathName = subfolderPathName
+                info.path = subfolderPath
+                info.reportCount = 1
+                subfoldersInfo[subfolderPath] = info
+            }
+        }else{
+            level = currentFolder.split('/').length - 1
+            displayName = entryAfterLevel(report, level)
+            let stats = entries[report];
+            reportParts.push('<a class="ui card" href="/result_listing/' + stats.report_name + '">' +
+                '  <div class="content">' +
+                '    <h1>' + displayName + '</h1>\n' +
+                '    <div class="meta">\n' +
+                '      <span class="date">Last ran ' + stats.time_diff + ' ago</span>\n' +
+                '    </div>' +
+                '    <span>\n' +
+                stats.count +
+                '        Runs\n' +
+                '    </span>' +
+                '<br/>' +
+                '    <span>\n' +
+                stats.scheduler_runs +
+                '        Scheduler Runs\n' +
+                '    </span>' +
+                '  </div>' +
+                '  <div class="extra content">' +
+                '      <span>Original report name: ' + stats.report_name + '</span>\n' +
+                '  </div>' +
+                '</a>');
+        }
+        
+        
+    }
+    // first add all folders
+    for (let subfolder in subfoldersInfo) {
+        info = subfoldersInfo[subfolder]
+        $cardContainer.append(
+            '<a class="ui card folder" href="/folder/' + info.path + '">' +
+            '  <div class="content">' +
+            '    <h1><i class="fa-solid fa-folder fa-xs"></i> ' + info.pathName + '</h1>\n' +
+            ' <span>Reports: ' + info.reportCount + '</span>' + 
+            '  </div>' +
+            '</a>');
+    }
+    // only then add individual items
+    for (let idx in reportParts) {
+        $cardContainer.append(reportParts[idx])
+    }
+}
+
+
 load_data = () => {
     $.ajax({
         url: `/core/get_all_templates_with_results`,
```

Comment:
I believe this is the URL which we should be parametrising with the "folder" - it will mean we can push the startsWith query down to storage.

Reply:
done